File: audit.js

const { chromium } = require('playwright');

(async () => {
  const browser = await chromium.launch();
  const ctx = await browser.newContext({ viewport: { width: 1280, height: 800 } });
  const page = await ctx.newPage();

  const errors = [];
  page.on('pageerror', e => errors.push('JS ERROR: ' + e.message));
  page.on('console', msg => {
    if (msg.type() === 'error') errors.push('CONSOLE: ' + msg.text());
  });

  const pages = [
    { url: 'http://127.0.0.1:8020/index.html', name: 'index' },
    { url: 'http://127.0.0.1:8020/condiciones.html', name: 'condiciones' },
    { url: 'http://127.0.0.1:8020/catalogo.html', name: 'catalogo' },
    { url: 'http://127.0.0.1:8020/contacto.html', name: 'contacto' }
  ];

  const results = [];

  for (const p of pages) {
    await page.goto(p.url, { waitUntil: 'networkidle' });
    await page.waitForTimeout(500);

    const screenshot = p.name + '.png';
    await page.screenshot({ path: 'C:/Users/HP/AppData/Local/hermes/cache/' + screenshot, fullPage: true });

    // Verificar problemas visuales
    const issues = await page.evaluate(() => {
      const issues = [];

      // 1. Texto blanco sobre fondo blanco
      const all = document.querySelectorAll('h1,h2,h3,h4,p,span,li,strong,a');
      let invisible = 0;
      all.forEach(el => {
        const cs = getComputedStyle(el);
        const c = cs.color;
        const bg = cs.backgroundColor;
        if (c.match(/255,\s*255,\s*255/) || c === 'white') {
          // esta en fondo blanco?
          let parent = el;
          let foundWhite = false;
          while (parent && parent !== document.body) {
            const pbg = getComputedStyle(parent).backgroundColor;
            if (pbg.match(/255,\s*255,\s*255/) || pbg === 'white') {
              foundWhite = true;
              break;
            }
            parent = parent.parentElement;
          }
          if (foundWhite) invisible++;
        }
      });

      // 2. Imagenes con object-position que cortan cabezas
      const photos = document.querySelectorAll('img');
      const totalImgs = photos.length;

      // 3. Nav tiene 4 links?
      const navLinks = document.querySelectorAll('.nav-links a').length;

      // 4. Footer tiene 4 columnas?
      const footerCols = document.querySelectorAll('.footer-col').length;

      // 5. Hay crema? (#F5EDE3)
      const hasCrema = document.documentElement.outerHTML.includes('#F5EDE3') ||
                       document.documentElement.outerHTML.includes('#f5ede3');

      // 6. Hay gris claro?
      const hasGray = document.documentElement.outerHTML.includes('#F6F6F7') ||
                      document.documentElement.outerHTML.includes('#f6f6f7');

      // 7. Hay Loreto?
      const hasLoreto = document.body.innerText.includes('Loreto');

      // 8. Hay Santa Barbara?
      const hasAddress = document.body.innerText.includes('Santa Bárbara');

      // 9. H1 y H2 existen y son visibles
      const h1 = document.querySelector('h1');
      const h2 = document.querySelector('h2');
      const h1Visible = h1 ? (h1.offsetHeight > 0) : false;

      return {
        invisibleTextCount: invisible,
        totalImages: totalImgs,
        navLinks,
        footerCols,
        hasCrema,
        hasGray,
        hasLoreto,
        hasAddress,
        h1Visible,
        pageHeight: document.body.scrollHeight
      };
    });

    results.push({ name: p.name, url: p.url, issues, errors });
  }

  await browser.close();
  console.log(JSON.stringify(results, null, 2));
})();

File: _audit.py

import asyncio, os, json
from playwright.async_api import async_playwright

async def main():
    async with async_playwright() as p:
        browser = await p.chromium.launch()
        ctx = await browser.new_context(viewport={"width": 1280, "height": 800})
        page = await ctx.new_page()

        pages = [
            ("http://127.0.0.1:8020/index.html", "index"),
            ("http://127.0.0.1:8020/condiciones.html", "condiciones"),
            ("http://127.0.0.1:8020/catalogo.html", "catalogo"),
            ("http://127.0.0.1:8020/contacto.html", "contacto"),
        ]

        results = []
        for url, name in pages:
            errors = []
            page.on("pageerror", lambda e: errors.append(str(e)))
            await page.goto(url, wait_until="networkidle")
            await page.wait_for_timeout(800)

            shot = os.path.join(r"C:\Users\HP\AppData\Local\hermes\cache\screenshots", f"audit_{name}.png")
            await page.screenshot(path=shot, full_page=True)

            issues = await page.evaluate("""() => {
              const out = {};
              out.title = document.title;
              out.h1Text = document.querySelector('h1')?.innerText;
              out.navLinks = document.querySelectorAll('.nav-links a').length;
              out.footerCols = document.querySelectorAll('.footer-col').length;
              out.totalImgs = document.querySelectorAll('img').length;
              out.hasCrema = document.documentElement.outerHTML.toLowerCase().includes('#f5ede3');
              out.hasGray = document.documentElement.outerHTML.toLowerCase().includes('#f6f6f7');
              out.hasSantaBarbara = document.body.innerText.includes('Santa Bárbara') || document.body.innerText.includes('Santa Barbara');
              out.hasLoreto = document.body.innerText.includes('Loreto');
              out.hasTurquesa = document.body.innerText.toLowerCase().includes('turquesa');
              out.hasRecoleta = document.body.innerText.includes('Recoleta');
              out.hasFaq = document.body.innerText.includes('FAQ') || document.body.innerText.includes('faqs');
              out.condCards = document.querySelectorAll('.cond-card').length;
              out.trajeCards = document.querySelectorAll('.traje-card-light, .traje-card').length;

              // Detectar texto blanco sobre fondo blanco
              const invisibles = [];
              document.querySelectorAll('h1,h2,h3,h4,p,span,li,strong,a').forEach(el => {
                const cs = getComputedStyle(el);
                if (cs.color.match(/255,\s*255,\s*255/) || cs.color === 'white') {
                  let parent = el;
                  let bgWhite = false;
                  while (parent && parent !== document.body) {
                    const pbg = getComputedStyle(parent).backgroundColor;
                    if (pbg.match(/255,\s*255,\s*255/) || pbg === 'white') {
                      bgWhite = true;
                      break;
                    }
                    parent = parent.parentElement;
                  }
                  if (bgWhite && el.innerText && el.innerText.trim().length > 0) {
                    invisibles.push(el.tagName + ': ' + el.innerText.substring(0, 50));
                  }
                }
              });
              out.invisibles = invisibles.slice(0, 5);
              out.invisibleCount = invisibles.length;
              return out;
            }""")
            issues['errors'] = errors
            results.append({"name": name, "url": url, **issues})

        await browser.close()
        print(json.dumps(results, indent=2, ensure_ascii=False))

asyncio.run(main())
